indexd/index/urls/v0.py

```diff
@@ -0,0 +1,115 @@
+import json
+
+from flask import Blueprint, request, Response
+
+from indexd.errors import UserError
+from indexd.index import request_args_to_params
+from indexd.index.drivers.query.urls import AlchemyURLsQueryDriver
+
+
+urls = Blueprint("urls", __name__)
+
+
+@urls.route("/q", methods=["GET"])
+@request_args_to_params
+def query(exclude=None, include=None, version=None, fields=None, limit=100, offset=0, **kwargs):
+    """ Queries indexes based on URLs
+    Args:
+        exclude (str): only include documents (did) with urls that does not match this pattern
+        include (str): only include documents (did) with a url matching this pattern
+        version (str): return only records with version number
+        fields (str): comma separated list of fields to return, if not specified return all fields
+        limit (str): max results to return
+        offset (str): where to start the next query from
+    Returns:
+        flask.Response: json list of matching entries
+            `
+                [
+                    {"did": "AAAA-BB", "rev": "1ADs" "urls": ["s3://some-randomly-awesome-url"]},
+                    {"did": "AAAA-CC", "rev": "2Nsf", "urls": ["s3://another-randomly-awesome-url"]}
+                ]
+            `
+    """
+    # parameter validation
+    if kwargs:
+        return Response("Unexpected Parameter(s) {}".format(kwargs), 400)
```

Comment:
don't need to use `Response` here; also this isn't consistent with the rest of the API, the error should be in JSON

Reply:
also—I guess it's fine this way—but consider if just ignoring extra args is fine or if erroring is necessary

Reply:
I'll see see to formatting this response as json, I do think the extra args makes sense, it shows the args received and typos can quickly be resolved